src/services/refactors/convertParamsToDestructuredObject.ts

```diff
@@ -51,7 +51,23 @@ namespace ts.refactor.convertParamsToDestructuredObject {
         changes: textChanges.ChangeTracker,
         functionDeclaration: ValidFunctionDeclaration,
         groupedReferences: GroupedReferences): void {
-        const newParamDeclaration = map(createNewParameters(functionDeclaration, program, host), param => getSynthesizedDeepClone(param));
+        const signature = groupedReferences.signature;
+        const newParamDeclaration = map(createNewParameters(signature ?? functionDeclaration, program, host), param => getSynthesizedDeepClone(param));

```

Comment:
I think you need to create parameters for both `signature` and `functionDeclaration`. It's possible for the function to be assignable to the signature without it being identical. And I'm not sure whether it's safe to replaceNode with the same node twice.

Reply:
Related, I'd like to see a test for non-identical assignability:

```ts
interface I { m(a) }
const o: I = { m(a,b) { } }
```

Also someday it should work for type aliases too, although if it doesn't work out-of-the-box don't worry about it for this PR.